src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      stateCopy = {};
+      result.push({});
+    } else if (actions[i].type === 'addProperties') {
```

Comment:
You can write all values in constant. 
const ACTION_CASES = {
        add: 'addProperties',
}